requirements.txt

```diff
@@ -14,9 +14,9 @@ types-requests==2.28.11.8
 types-setuptools==67.1.0.0
 
 # linting
-black[jupyter]==22.12.0
+black[jupyter]==23.7.0
 mypy==0.991
 pre-commit==3.2.0
 pylint==2.15.10
-ruff==0.0.259
+ruff==0.0.285
```

Comment:
I'm slightly hesitant to update the ruff version. Can you explain why it's needed?

Reply:
it covers more cases